packages/datadog-plugin-rhea/src/index.js

```diff
@@ -1,73 +1,92 @@
+'use strict'
+
+let CircularBuffer
+let popIf
+
+function patchCircularBuffer (CircularBuffer) {
+  popIf = CircularBuffer.prototype.pop_if
+  CircularBuffer.prototype.pop_if = function (fn) {
+    const wrappedFn = entry => {
+      const shouldPop = fn(entry)
+      if (shouldPop && entry._dd) {
+        const state = entry.remote_state ? entry.remote_state.constructor.composite_type : 'accepted'
+        finish(entry, state)
+      }
+      return shouldPop
+    }
+    return popIf.call(this, wrappedFn)
+  }
+}
+
 function createWrapSend (tracer, config) {
   return function wrapSend (send) {
     return function sendWithTrace (msg, tag, format) {
-      const name = this.options.target && this.options.target.address
+      const name = this.options && this.options.target && this.options.target.address
         ? this.options.target.address : 'amq.topic'
-      return tracer.trace('rhea.sender.send', { tags: {
-        'resource.name': name,
-        'service.name': config.service || `${tracer._service}-amqp`,
-        'span.kind': 'producer',
-        'amqp.link.target.address': name,
-        'amqp.link.role': 'sender',
-        'out.host': this.connection.options.host,
-        'out.port': this.connection.options.port
-      } }, (span, done) => {
-        msg.delivery_annotations = msg.delivery_annotations || {}
-        tracer.inject(span, 'text_map', msg.delivery_annotations)
-        const delivery = send.apply(this, arguments)
-        if (this.options.snd_settle_mode !== 1) {
-          delivery._dd = { done, span }
-        } else {
-          done()
+      let host
```

Comment:
Can the host/port part be extracted to a function? This is starting to get a bit verbose.